modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/MavenUtilsForArtifact.java

```diff
@@ -0,0 +1,79 @@
+
+import static org.mule.runtime.api.util.Preconditions.checkState;
+import static org.mule.runtime.module.artifact.api.descriptor.DeployableArtifactDescriptor.MULE_POM_PROPERTIES;
+
+import static java.io.File.separator;
+import static java.lang.String.format;
+
+import static org.apache.commons.io.filefilter.DirectoryFileFilter.DIRECTORY;
+
+import org.mule.runtime.core.api.util.PropertiesUtils;
+import org.mule.runtime.module.artifact.api.descriptor.ArtifactDescriptorCreateException;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * Provides utility methods for artifact-activation module to work with Maven
+ */
+public class MavenUtilsForArtifact {
+
+  /**
+   * Returns the {@link Properties} with the content of {@code pom.properties} from a given artifact folder.
+   *
+   * @param artifactFolder folder containing the exploded artifact file.
+   */
+  public static Properties getPomPropertiesFolder(File artifactFolder) {
+    final File artifactPomProperties = lookupPomPropertiesMavenLocation(artifactFolder);
+    Properties properties;
+    try (InputStream artifactPomPropertiesInputStream = new FileInputStream(artifactPomProperties)) {
+      properties = PropertiesUtils.loadProperties(artifactPomPropertiesInputStream);
+    } catch (IOException e) {
+      throw new ArtifactDescriptorCreateException(format("There was an issue reading '%s' for the artifact '%s'",
+                                                         artifactPomProperties.getName(), artifactFolder.getAbsolutePath()),
+                                                  e);
+    }
+    return properties;
+  }
+
+  /**
+   * Look up pom properties.
+   */
+  public static File lookupPomPropertiesMavenLocation(File artifactFolder) {
+    File mulePropertiesPom = null;
+    File lookupFolder = new File(artifactFolder, "META-INF" + separator + "maven");
+    while (lookupFolder != null && lookupFolder.exists()) {
+      File possiblePomPropertiesLocation = new File(lookupFolder, MULE_POM_PROPERTIES);
+      if (possiblePomPropertiesLocation.exists()) {
+        mulePropertiesPom = possiblePomPropertiesLocation;
+        break;
+      }
+      File[] directories = lookupFolder.listFiles((FileFilter) DIRECTORY);
+      checkState(directories != null || directories.length == 0,
+                 format("No directories under %s so pom.properties file for artifact in folder %s could not be found",
+                        lookupFolder.getAbsolutePath(), artifactFolder.getAbsolutePath()));
+      checkState(directories.length == 1,
+                 format("More than one directory under %s so pom.properties file for artifact in folder %s could not be found",
+                        lookupFolder.getAbsolutePath(), artifactFolder.getAbsolutePath()));
+      lookupFolder = directories[0];
```

Comment:
why is this and the while needed? In what cases can the properties be nested in a lower level?